### PR TITLE
Add Digital Stream Writer tests

### PR DESCRIPTION
**Pull request by zhindes**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This ensures we have a test per method of the Digital stream writer classes. These tests use digital loopback to validate that the data made it all the way through the Python/C bindings correctly to the underlying driver.

other miscellany
* better workaround for #392
* more pythonic code for initializing a list of a single value

### Why should this Pull Request be merged?

This will give us confidence that we haven't broken anything as we refactor parts of the Python / C bindings.

### What testing has been done?

```
> poetry run ni-python-styleguide fix tests && poetry run ni-python-styleguide lint tests && poetry run pytest -rs -k test_stream_writers_do
===================================== test session starts ======================================
platform win32 -- Python 3.9.13, pytest-8.0.2, pluggy-1.4.0
rootdir: C:\dev\nidaqmx-python
configfile: pyproject.toml
testpaths: tests
plugins: cov-4.1.0, mock-3.12.0
collected 1951 items / 1897 deselected / 54 selected

tests\component\test_stream_writers_do.py ............................................... [ 87%]
.......                                                                                   [100%]

============================= 54 passed, 1897 deselected in 18.00s =============================
```

_(I also checked `test_stream_writers_ao` and `test_stream_readers_di` for the minor fixes there.)_

### CODE REVIEW

File: tests/component/test_stream_writers_do.py [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+    return task
+
+
+@pytest.fixture
+def do_single_channel_multi_line_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    chan = task.do_channels.add_do_chan(
+        flatten_channel_string(real_x_series_device.do_lines[:8].name),
```
Comment (bkeryan) [inline tests/component/test_stream_writers_do.py:47]: `PhysicalChannel.name` should be a str, not a list. You are hitting this bug: https://github.com/ni/nidaqmx-python/issues/392

As a workaround, I suggest `flatten_channel_string(real_x_series_device.do_lines.channel_names[:8])`

Or you could fix the bug 😁

File: tests/component/test_stream_writers_do.py [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    _start_do_task(task, is_port=True, num_chans=task.number_of_channels)
+    return task
+
+
+def _start_di_task(task: nidaqmx.Task) -> None:
+    # Don't reserve the lines, so we can read what DO is writing.
+    for chan in task.di_channels:
+        chan.di_tristate = False
```
Comment (bkeryan) [inline tests/component/test_stream_writers_do.py:118]: You don't have to iterate through channels. Use `.all` to access all channels together.
```suggestion
    task.di_channels.all.di_tristate = False
```

File: tests/component/test_stream_writers_do.py [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+
+
+def _start_do_task(task: nidaqmx.Task, is_port: bool = False, num_chans: int = 1) -> None:
+    # We'll be doing on-demand, so start the task and drive all lines low
+    task.start()
+    if is_port:
+        if num_chans == 8:
+            task.write(0)
+        else:
+            task.write([0 for _ in range(num_chans)])
```
Comment (bkeryan) [inline tests/component/test_stream_writers_do.py:21]: FYI, I keep forgetting that list has an operator for this: `[0] * num_chans`

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: tests/component/test_stream_writers_do.py (hunk 1 above)
Comment (zhindes) [reply tests/component/test_stream_writers_do.py:47]: I'll workaround for now, and then fix that once I'm done with my set of new component tests

**Pull request merged**